app/frontend/src/cljs/teet/admin/indexes_view.cljs

```diff
@@ -0,0 +1,343 @@
+(ns teet.admin.indexes-view
+  "Indexes admin user interface"
+  (:require [teet.ui.material-ui :refer [Table TableHead TableBody TableRow TableCell Paper Collapse]]
+            [teet.ui.text-field :refer [TextField]]
+            [teet.ui.buttons :as buttons]
+            [herb.core :refer [<class]]
+            [clojure.string :as str]
+            [teet.ui.icons :as icons]
+            [teet.localization :refer [tr tr-enum]]
+            [teet.ui.form :as form]
+            [teet.ui.format :as format]
+            [teet.ui.typography :as typography]
+            teet.user.user-spec
+            [teet.ui.select :as select]
+            [teet.ui.util :refer [mapc]]
+            [teet.ui.query :as query]
+            [reagent.core :as r]
+            [teet.theme.theme-colors :as theme-colors]
+            [teet.common.common-styles :as common-styles]
+            [teet.ui.common :as common-ui]
+            [teet.ui.context :as context]
+            [teet.user.user-model :as user-model]
+            [teet.util.date :as date]
+            [teet.ui.date-picker :as date-picker]
+            [teet.admin.admin-controller :as admin-controller]
+            [teet.ui.url :as url]
+            [cljs-time.core :as t]
+            [teet.util.coerce :as uc]))
+
+(defn- by-month [start-date]
+  (iterate #(t/plus % (t/months 1)) start-date))
+
+(def enabled-indexes
+  #{
+    {:name "Bitumen index"
+     :id "bitumen"
+     :type :cost-index.type/bitumen-index
+     :unit "€"}
+    {
+     :name "Consumer Price index"
+     :id "consumer"
+     :type :cost-index.type/consumer-price-index
+     :unit "p"
+     }
+    })
+(defn- is-valid-index-id?
+  "Checks if id is defined in the 'enabled-indexes' variable and returns the index"
+  [index-id]
+  (filterv (fn [x] (if (= (:id x) index-id) x nil)) enabled-indexes)
+  )
+
+(defn index-save-form-button
+  [invalid-attributes {:keys [validate]}]
+  [:div {:class (<class common-styles/margin-left 1)}
+   [buttons/button-primary
+    {:on-click validate
+     :type :submit
+     :disabled (boolean (seq invalid-attributes))}
+    [:span (tr [:buttons :save])]]])
+
+(defn index-save-form-button*
+  []
+  (context/consume
+    :form
+    (fn [{:keys [invalid-attributes value footer]}]
+      [index-save-form-button @invalid-attributes value (update footer :validate
```

Comment:
why does this form need a specialized save button? should use the same form footer we have commonly, or document why it isn't working for this

Reply:
I understood this is required for the field validaton, but will try to get rid of it.